src/main/java/org/gradle/java/JigsawPlugin.java

```diff
@@ -56,7 +58,9 @@
     @Override
     public void apply(Project project) {
         LOGGER.debug("Applying JigsawPlugin to " + project.getName());
-
+        if (!project.getPlugins().hasPlugin(JAVA_PLUGIN) && !project.getPlugins().hasPlugin(JAVA_LIBRARY_PLUGIN)) {
+            project.getPluginManager().apply(JavaLibraryPlugin.class);
```

Comment:
Should apply the `JavaPlugin` instead, since the `JavaLibraryPlugin` already applies it.